feat: Update Karpenter pattern to use `v0.32.1` with `v1beta1` APIs (#1819)
aws-ia/terraform-aws-eks-blueprints: patterns/karpenter/README.md

## Validate

1. Test by listing the nodes in the cluster. You should see four Fargate nodes in the cluster:

    ```sh
    kubectl get nodes

    NAME                                                STATUS   ROLES    AGE     VERSION
    fargate-ip-10-0-11-195.us-west-2.compute.internal   Ready    <none>   5m20s   v1.28.2-eks-f8587cb
    fargate-ip-10-0-27-183.us-west-2.compute.internal   Ready    <none>   5m2s    v1.28.2-eks-f8587cb
    fargate-ip-10-0-4-169.us-west-2.compute.internal    Ready    <none>   5m3s    v1.28.2-eks-f8587cb
    fargate-ip-10-0-44-106.us-west-2.compute.internal   Ready    <none>   5m12s   v1.28.2-eks-f8587cb
    ```

2. Provision the Karpenter `EC2NodeClass` and `NodePool` resources which provide Karpenter the necessary configurations to provision EC2 resources:

    ```sh
    kubectl apply -f karpenter.yaml
    ```

> 3. With the Karpenter resources are in place, when pods are pending in the scheduler's queue, Karpenter will provision the necessary EC2 resources to satisfy those requests. You can demonstrate this with the example deployment provided. First deploy the example deployment which has the initial number replicas set to 0:

Author: ```suggestion
3. Once the Karpenter resources are in place, Karpenter will provision the necessary EC2 resources to satisfy any pending pods in the scheduler's queue. You can demonstrate this with the example deployment provided. First deploy the example deployment which has the initial number replicas set to 0:
```
